### ISSUE TITLE
User suggestion: add support for displaying angles [0°, 360°)

### ISSUE THREAD

**Issue by arouinfar** [phetsims/vector-addition#276]
A user wrote into phethelp:

> Vector addition PHET shows negative angles rather than positive angles in quadrants 3 and 4 of Cartesian plane when the “show theta” button is pressed.  It would be better for student understanding if the angle (counterclockwise from East) could take any value from 0 to 359 degrees. 
![image](https://user-images.githubusercontent.com/8419308/137387466-47135169-8e46-43fc-8ad7-009850cccaad.png)

Currently, the angle ranges (-180°, 180°] which is an appropriate choice. However, always defining the angle counterclockwise from the positive x-axis could be preferable for some users. 

I would consider adding support displaying the angle as [0°, 360°) via query parameter, but we can defer further discussion until this sim is revisited.

**Comment by pixelzoom** [comment]
In addition to a query parameter, also consider a Preferences control.

**Comment by matthew-blackman** [comment]
I agree that this would be a useful feature to add and has clear pedagogical value, especially for students learning to interpret angles in different contexts. I support the idea of including this as a Simulation Preferences radio control, as @pixelzoom suggested.

Co-assigning @kathy-phet to confirm whether this is within scope for the current round of work being done on the Vector Addition sim (PhET-iO instrumentation, Alt Input, UI Sound, Tier 1 Description).

**Comment by pixelzoom** [comment]
Here's what I need in order to continue:

* Name, 2 values, and default value for the Property<Range>.
* Name, 2 values, and default value for the query parameter.
* Mockup for the Preferences control: label, description, and labels for the 2 choices.


In a Google search, I see [0,360) referred to as "full rotation", while (-180,180] is referred to as "principal angle" or "angle range for heading".

**Comment by pixelzoom** [comment]
It will important to make this preference affect the view only.  To make the model switch between angle ranges, we would need to touch more code.  It could certainly be more destabilizing.  And it would likely break the PhET-iO API.

**Comment by pixelzoom** [comment]
The preference is implemented, but not wired into the view yet. It is a starting point for discussion.

Underlying the implementation is type `AngleConvention`:

```
export const AngleConventionValues = [
  'fullRotation',  // (0,360]
  'principalAngle' // [-180,180)
] as const;
export type AngleConvention = ( typeof AngleConventionValues )[number];
```

The query parameter is `?angleConvention`, whose values are `AngleConventionValues`, with default 'fullRotation'.

The Property is `angleConventionProperty: StringUnion<AngleConvention>`, and its initial value comes from the query parameter.

In Preferences > Simulation, the control looks like the screenshot below. (I'm using the look provided by PreferencesControl. I hate the layout, but that's what PhET decided on.)

![Image](https://github.com/user-attachments/assets/c5c9e292-90ec-4c12-b8d6-27ae4bf34676)

**Comment by pixelzoom** [comment]
Next step is for @matthew-blackman to review, before I wire this up to the display of angles in the view.

**Comment by matthew-blackman** [comment]
Thanks, @pixelzoom — this is looking great and definitely headed in the right direction.

A few thoughts and suggestions:

- Property Name: I’m fine with `angleConventionProperty`, but I’d recommend considering `angleDisplayConventionProperty` to clarify that this setting only affects the view and not the underlying model or data values (as you noted). While a bit more verbose, I think the clarity is worthwhile. That said, if we feel this makes the query parameter too long, I’m okay with `angleConventionProperty`.

- Option Labels / Terminology: For the values, I suggest using `signed` and `unsigned`. These terms appear to be both widely used and pedagogically clear — see [Wikipedia’s entry on signed angles](https://en.wikipedia.org/wiki/Angle#Signed_angles) as one reference.

```
signed → (−180°, 180°]

unsigned → [0°, 360°)
```

- Default Value: I support keeping the default as `signed` since this matches the currently published behavior of the sim.

- Query Parameter: Consistency with the property name — e.g., `angleConvention=signed` or `angleDisplayConvention=signed`. I personally don’t find `angleDisplayConvention` too long, but am open to whatever feels most aligned with our existing conventions.

- Preferences UI Labeling: The control label should probably match the property name — e.g., Angle Display Convention. For the radio button labels, I recommend:

Signed (−180°, 180°]

Unsigned [0°, 360°)

- Preferences UI Layout: I agree that the Preferences layout feels a bit awkward here. I’ve attached mockups showing what I get by updating the strings — curious if you have any suggestions for cleaning up the spacing or alignment. Any refinements that make the UI feel less cramped would be an improvement.

<img width="1388" alt="Image" src="https://github.com/user-attachments/assets/683b699c-1598-406a-b45f-965a234b3232" />


@pixelzoom hope this provides some clarity. Let me know your thoughts!

**Comment by pixelzoom** [comment]
@matthew-blackman and I met about this on Zoom, and I made related changes in the above commits.  Summary:

* Did not change the names `angleConvention` and `angleConventionProperty` because the Preferences control label is "Angle Convention", and "Angle Display Convention" would not improve the UX. 
* Changed values to "signed" and "unsigned", with default "signed".
* Revised the layout and labeling of the Preferences control to look like the screenshot below. It's still kind of ugly, but that's a common-code problem.

<img width="476" alt="Image" src="https://github.com/user-attachments/assets/bacf7195-eed2-48e3-9de9-3773bd19789c" />

**Comment by pixelzoom** [comment]
Radio button labels were localized in the above commits.

Next step is for me to wire this preference into the view, so that it affects the display of angle values.

**Comment by pixelzoom** [comment]
> Note to self: angle display is handled by `VectorAngleNode`.

Angle is also displayed in `VectorValuesNumberDisplay` (and seems to be missing the degrees units), for example:

<img width="664" alt="Image" src="https://github.com/user-attachments/assets/c4916a94-846c-4699-8be7-28d233357040" />

**Comment by pixelzoom** [comment]
Our angle ranges are intended to have an inclusive and exclusive min/max, as indicated by the notation in Preferences:

![Image](https://github.com/user-attachments/assets/f2220c6d-0e77-4a1f-9f53-69e7fb813d03)

This seems to be the case in the first 3 screens.  But in the Equations screen, I see that it's possible to set angle to -180 and +180, as in the screenshots below.  I'm guessing that the NumberPicker range should be [-179,180] or [0,359], depending on the Preferences setting.  @matthew-blackman does that sound correct to you?

![Image](https://github.com/user-attachments/assets/6f64df86-1b71-477d-88f4-e6f46e7f9701)

![Image](https://github.com/user-attachments/assets/e1de9bc4-fba9-480b-af6e-b5ef96b14316)

**Comment by matthew-blackman** [comment]
Thanks for flagging this. I agree that on the Explore 2D and Lab screens—where users manipulate vectors by dragging—the current behavior of defaulting to 0° instead of 360°, and +180° instead of -180°, feels fine and intuitive.

However, for the Equations screen, I think there's value in allowing users to set angles to the full range of -180 to 180 or 0 to 360 using the NumberPickers:

- Since the pickers increment by 5°, enforcing an exclusive upper/lower bound would restrict users to -175° or 355°, which feels arbitrary and potentially confusing.
- It’s pedagogically useful for students to see that -180° and 180° (or 0° and 360°) represent the same direction. Preventing selection of those values would obscure that.
- I don't see an issue with allowing these endpoints on the Equations screen, even if other screens wrap. The interaction modes are different enough that this doesn’t seem inconsistent.
- I’d also suggest simplifying the Preferences UI by labeling the options as “-180 to 180” and “0 to 360” without relying on inclusive/exclusive bracket notation, which isn’t widely understood and may not matter much to users.

Overall, I think wrapping the angle display on the Explore 2D and Lab screens, while allowing both -180 and +180, or 0 and 360 on the Equations screen, creates the best of all worlds for the user experience manipulating and understanding vectors on these screens.

**Comment by pixelzoom** [comment]
What's making it difficult to wire this Preference into the sim is that I cannot locate where the angle range is being constrained to [-180,180], and why it's exclusive of -180 on the first 3 screens and not on the Equations screen.  When I locate that, I'm going to document it in the code, and model.md or implementation-notes.md.

**Comment by pixelzoom** [comment]
Ugh... One more thing to do.  The arrow that denotes the angle needs to be different for the 2 conventions. 

For example, this is correct for -180 to 180:

<img width="196" alt="Image" src="https://github.com/user-attachments/assets/0ea9046a-c92e-4e69-8654-3ef37e8df0bc" />

But this is incorrect for 0 to 360. In this case, the angle arrow should always follow a counterclockwise path:

<img width="204" alt="Image" src="https://github.com/user-attachments/assets/ae5080b0-0b45-4e02-8680-96999e480cce" />

**Comment by pixelzoom** [comment]
Pickers have immutable ranges.  So in the Base Vectors accordion box, it was necessary to create 2 pickers (signed and unsigned) and show the one that matches the selected angle convention.  So while the angle is shown with the correct convention throughout the UI, there's still some PhET-iO work to do.  Specifically:

- [ ] Revisit PhET-iO instrumentation for `dAnglePicker` and `eAnglePicker`.

**Comment by pixelzoom** [comment]
@matthew-blackman FYI, I've added the degree symbol for angles shown on the graph.

<img width="227" alt="Image" src="https://github.com/user-attachments/assets/c515c344-c3a0-4c81-ae27-624e1057299e" />

**Comment by matthew-blackman** [comment]
This is working great for both preference options, and switching between them converts as expected.

I agree with @pixelzoom that this is not a problem right now, and we can hold off until user/client request:
> there is no API for decrementInputListener.firedEmitter and incrementInputListener.firedEmitter

Looks excellent @pixelzoom! This is a great feature to have added to the sim. I'll make sure to ask QA to do some additional testing with this in the next release cycle. This issue can be closed.
